From: David Sterba <dsterba@suse.cz>
Date: Fri, 23 Jan 2015 18:43:31 +0100
Patch-mainline: 4.7
Git-commit: ee6111386a1b304f8bf589d36810d53e3b27ee20
References: bsc#1012452
Subject: [PATCH] btrfs: add read-only check to sysfs handler of
 features

We don't want to trigger the change on a read-only filesystem, similar
to what the label handler does.

Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/sysfs.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/btrfs/sysfs.c
+++ b/fs/btrfs/sysfs.c
@@ -120,6 +120,9 @@ static ssize_t btrfs_feature_attr_store(
 	if (!fs_info)
 		return -EPERM;
 
+	if (fs_info->sb->s_flags & MS_RDONLY)
+		return -EROFS;
+
 	ret = kstrtoul(skip_spaces(buf), 0, &val);
 	if (ret)
 		return ret;
